doc/sql.extensions/README.MED.md

```diff
@@ -0,0 +1,334 @@
+```
+Plugin = JDBC {
+    Module = $(dir_plugins)/jdbc_provider
+    Config = JDBC_config
+}
+
+Config = JDBC_config {
+    Dir = $(this)/jdbc
+}
+```
+To connect to Firebird databases, the Engine and Remote providers are available by default and require no additional definition in `plugins.conf`.
+
+### Foreign server
+
+A foreign server is a logical representation of a remote data source inside the DBMS. The DBMS connects to the foreign server to execute queries. The foreign server provides metadata and connection information about the remote data source. Usually, the external server stores the connection parameters such as connection string, port, database type, etc.
+
+#### CREATE SERVER definition
+
+```sql
+CREATE SERVER [IF NOT EXISTS] <foreign server name>
+        [{FOREIGN DATA WRAPPER | USING PLUGIN} <provider (plugin)>] [OPTIONS (<option> [,<option> ...] )]
+    <option> ::= {
+        CONNECTION_STRING = '<connection string>'
+        | USER = '<user name>'
+        | PASSWORD = '<password>'
+        | ROLE = '<role>'
+        | <option name> = 'value'
+        | <option name> [{FILE | ENV}] 'value' }
+```
+The `SERVER` clause specifies the name of the server.
+The `FOREIGN DATA WRAPPER` or `USING PLUGIN` clause specifies the provider to use to access foreign data. If no clause is specified, the providers specified in the Providers parameter of the `firebird.conf` configuration file will be tried for connection. The clause `USING PLUGIN` is not specified in the standard, but it is familiar to Firebird developers and users, so it is proposed as an extension to the standard definition of a foreign server.
+The optional `OPTIONS` clause defines additional connection parameters to the foreign server. The required connection parameters depend on the provider \(plugin\) used to access the data. The option value can be read from a file or an environment variable. If it is necessary to read the value from a file, then `FILE` is specified after the option name or `ENV` is specified to read from the environment variable. The server should have the permissions to read the specified file. General options are:
+* `CONNECTION_STRING` - connection string to the foreign server \(depends on the foreign server specification\), if the option is not specified, an attempt will be made to connect to the current database;
+* `USER` - the user on whose name the connection will be established;
+* `PASSWORD` - user password;
+* `ROLE` - the role with which the specified user will connect to the foreign server.
+
+The `CREATE SERVER` statement can be executed by administrators or users with the `CREATE SERVER` privilege.
+
+#### ALTER SERVER definition
+
+```sql
+ALTER SERVER <foreign server name>
+        [[DROP] FOREIGN DATA WRAPPER | [DROP] USING PLUGIN <provider(plugin)>] [OPTIONS(<option> [, <option> ...] )]
+    <option> ::= {
+        [DROP] <option name> [= 'value']
+        | [DROP] <option name> [{FILE | ENV}] ['value'] }
```

Comment:
This description confuses people because it's unclear what's required and what's optional. A better explanation would be:

```
ALTER SERVER <foreign server name>
        [[DROP] {FOREIGN DATA WRAPPER | USING PLUGIN} <provider(plugin)>] 
        [OPTIONS (<option> [, <option> ...] )]

<option> ::= {
          <option name> = 'value'
        | <option name> [{FILE | ENV}] 'value' 
        | DROP <option name>
}
```